Slack-Apps/week3/homework/responses/3.1-reaction-event-listener-and-threads.md

```diff
@@ -0,0 +1,49 @@
+# Google Translate API: Event Listener & Threads
+
+During the livestream, you were introduced to two new concepts: a new event listener to emoji reactions and a new way of having your Slack App respond to a thread. In this issue, you will be incorportating these concepts as you integrate your Slack App with your first third party API, Google Translate!
+
+## Event Listeners
+
+Before we introduce the Google Translate API, make sure you understand how the new event listener works as well as how to make your Slack App respond to a thread:
+
+Emoji Reactions: https://api.slack.com/events/reaction_added
+ ```javascript 
+ app.event('reaction_added', async ({ event, client }) => {
+ /* Listening for Reaction Emojis */
+ });
+
+```
+Using ```app.event('reaction_added')```acts as an event listener whenever a user responds to a message with a reaction (this can be any emoji ex: smile, heart, or even a flag!). We will be using the reaction_added event listener for our Slack App that translates a user's message to a language based on the nationality of the flag the user reacts to the message with.
+
+```javascript
+const { reaction, user, item: { channel, ts } } = event;
+```
+
+Here, we are setting the reaction value to a constant in order to compare it to a list of languages. However, the names of some emojis don't exactly represent the language we need to translate to which is why we have to create a map of country codes corresponding with the APIs values. An example of this:
```

Comment:
Just in case, tell them what emoji names you're referring to. (Tell them to search "flag" on Slack and they can see that :flag-au: corresponds to the Australian flag)

Reply:
Would also be helpful if you showed examples.
> Ex: Name of [flag name] is "[name value]" which is not a value the Google Translate API incorporates